lib/seahorn/BvOpSem2FatMemMgr.cc

```diff
@@ -224,15 +278,27 @@ class FatMemManager : public OpSemMemManager {
   FatMemValTy storeIntToMem(Expr _val, FatPtrTy ptr, FatMemValTy mem,
                             unsigned byteSz, uint64_t align) override {
     return mkFatMem(
-        m_mem.storeIntToMem(_val, mkRawPtr(ptr), mkRawMem(mem), byteSz, align));
+        m_main.storeIntToMem(_val, mkRawPtr(ptr), mkRawMem(mem), byteSz, align),
+        m_main.storeIntToMem(0, mkRawPtr(ptr), mkSlot0Mem(mem), byteSz, align),
+        m_main.storeIntToMem(0, mkRawPtr(ptr), mkSlot1Mem(mem), byteSz, align));
   }
 
   /// \brief Stores a pointer into memory
   /// \sa storeIntToMem
   FatMemValTy storePtrToMem(FatPtrTy val, FatPtrTy ptr, FatMemValTy mem,
                             unsigned byteSz, uint64_t align) override {
-    return mkFatMem(m_mem.storePtrToMem(mkRawPtr(val), mkRawPtr(ptr),
-                                        mkRawMem(mem), byteSz, align));
+    // assert that ptr is FatPtrTy
+    MemValTy main = m_main.storePtrToMem(mkRawPtr(val), mkRawPtr(ptr),
+                                         mkRawMem(mem), byteSz, align);
+    MemValTy slot0 = m_slot0.storeIntToMem(getFatData(val, 0), mkRawPtr(ptr),
```

Comment:
not storing `byteSz`. See comment in `storeIntToMem`

Reply:
I am not seeing why `byteSz` is wrong.